CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+cmake_minimum_required(VERSION 3.10.2)
+
+# where to find our CMake modules
+set(CMAKE_MODULE_PATH "$ENV{CARMA_OPT_DIR}/cmake")
+# Required for std::lock and syntax
+set(CMAKE_CXX_STANDARD 17)
+
+# Needed so CMake can find the vendored PROJ4 module file. Th FindPROJ4.cmake
+# module file can be removed if we upgrade to a more recent PROJ version. See
+# https://github.com/usdot-fhwa-stol/carma-platform/issues/2139 for updates.
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+project(carma-geodetic
+    DESCRIPTION "CARMA geodetic library"
+    HOMEPAGE_URL https://github.com/usdot-fhwa-stol/carma-geodetic-lib
+	VERSION 0.0.1
+    LANGUAGES CXX
+)
+
+find_package(Threads)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
+include(cmake/dependencies.cmake)
+
+add_library(${PROJECT_NAME} SHARED)
```

Comment:
Coupling the project name to the library name is a frowned-upon practice. It also makes reading the CMake configurations more difficult. This and all references to ${PROJECT_NAME} should be replaced with the intended target name, even if it matches the project's name.

Reply:
Hmmm i have seen a lot of projects do this including the units library you included. Could you maybe explain more about why it is important to avoid using ${PROJECT_NAME} in this way. Since this project is essentially just this library I do not see why this would be difficult to read or a concern.

Reply:
Here is an excerpt from Craig Scott's _Professional CMake: A Practical Guide_, Section 4.6. Recommended Practices:

> Target names need not be related to the project name. It is common to see tutorials and examples use a variable for the project name and reuse that variable for the name of an executable [in our case, library] target:
>
> [example CMake snippet removed]
>
> This only works for the most basic of projects and encourages a number of bad habits. Consider the project name and executable [or library] name as being separate, even if initially they start out the same. Set the project name directly rather than via a variable, choose a target name according to what the target does rather than the project it is part of an assume the project will eventually need to define more than one target. This reinforces better habits which will be important when working on more complex multi-target projects [for example, ROS packages that define both libraries and executables].

Reply:
I am ok with changing this to enforce good habits for defining CMakeLists.txt files. Want to make sure I understand the concern here. The concern is that defining targets with project name links the target to the project which should ideally be 2 independent concepts since more complex CMake files can have multiple targets. 

In this case for stol-apt repos my understanding is that ideally, these would always be single target projects  since any dependent targets should probably be independent repos pulled in via apt stol debian package repository rather than targets included in one larger monolithic CMake project right? Or do you see utility in having mutli-target repos here that get build in to debian package?

Reply:
Packaging is an orthogonal concept to project structure. A good project structure gives packagers the freedom to package things how they see fit.

Multi-target projects are essentially inherent to how we're developing. For example, a library is one target, and the unit tests are another. So in this case we already have multiple targets. I'm not sure if that answers your question.

Reply:
If we want to address this I suggest we create a separate issue. I think most of the cmake files from the carma-builds image also depend on target which works out well in practice right now for the libraries that use  carma-builds since they just build a single project executable and test executable. I can see that this separation could be helpful if in the future other projects injest our libraries using the source code and CMake files although currently that is not really the intent of these libraries so doing work to enable this may be low priority for now.

Reply:
I see what you mean. The carma-builds CMake modules all assume the targets use ${PROJECT_NAME} as their name, so we are forced to follow that in this case. Let's open a new issue in the [carma-builds](https://github.com/usdot-fhwa-stol/carma-builds) repo about improving their CMake modules.